### PR TITLE
Creates random passwords for additional_users

### PR DESCRIPTION
**Pull request by samcre**
For every Cloud SQL instance type, if `additional_users.password` it's
an empty string, the module will create a random password.

Also, adds an output to retrieve users and passwords from other
resources.

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_documentation
 The documentation needs to be regenerated. Please run `make generate_docs`. 
 ```bash 
Checking for documentation generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' '--exclude=autogen' '--exclude=*.tfvars' /workspace/modules/mssql/README.md /tmp/tmp.4rdt9tqi1E/generate_docs/workspace/modules/mssql/README.md
51a52
> | additional\_users | Map of additional users and passwords |
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' '--exclude=autogen' '--exclude=*.tfvars' /workspace/modules/mysql/README.md /tmp/tmp.4rdt9tqi1E/generate_docs/workspace/modules/mysql/README.md
54a55
> | additional\_users | Map of additional users and passwords |
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' '--exclude=autogen' '--exclude=*.tfvars' /workspace/modules/postgresql/README.md /tmp/tmp.4rdt9tqi1E/generate_docs/workspace/modules/postgresql/README.md
54a55
> | additional\_users | Map of additional users and passwords |
Error: Documentation generation has not been run, please run the
'make docker_generate_docs' command and commit the above changes.
 ``` 
- ⚠️check_terraform
 Failed Terraform check. More details below. 
 ```bash 
Running terraform fmt
Running terraform validate
terraform_validate ./examples/mssql-public 
Error: 
Missing resource instance key
  on ../../modules/mssql/main.tf line 178, in resource "google_sql_user" "additional_users":
 178:   password   = lookup(each.value, "password", 
random_password.additional_passwords
.result)
Because random_password.additional_passwords has "for_each" set, its
attributes must be accessed on specific instances.
For example, to correlate with indices of a referring resource, use:
    random_password.additional_passwords[each.key]
terraform_validate ./examples/mysql-ha 
Success!
 The configuration is valid.
terraform_validate ./examples/mysql-private 
Success!
 The configuration is valid.
terraform_validate ./examples/mysql-public 
Success!
 The configuration is valid.
terraform_validate ./examples/postgresql-ha 
Success!
 The configuration is valid.
terraform_validate ./examples/postgresql-public 
Success!
 The configuration is valid.
terraform_validate ./examples/postgresql-public-iam 
Success!
 The configuration is valid.
terraform_validate ./modules/mssql 
Error: 
Missing resource instance key
  on main.tf line 178, in resource "google_sql_user" "additional_users":
 178:   password   = lookup(each.value, "password", 
random_password.additional_passwords
.result)
Because random_password.additional_passwords has "for_each" set, its
attributes must be accessed on specific instances.
For example, to correlate with indices of a referring resource, use:
    random_password.additional_passwords[each.key]
terraform_validate ./modules/mysql 
Success!
 The configuration is valid.
terraform_validate ./modules/postgresql 
Success!
 The configuration is valid.
terraform_validate ./modules/private_service_access 
Success!
 The configuration is valid.
terraform_validate ./modules/safer_mysql 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/mssql-ha 
Error: 
Missing resource instance key
  on ../../../modules/mssql/main.tf line 178, in resource "google_sql_user" "additional_users":
 178:   password   = lookup(each.value, "password", 
random_password.additional_passwords
.result)
Because random_password.additional_passwords has "for_each" set, its
attributes must be accessed on specific instances.
For example, to correlate with indices of a referring resource, use:
    random_password.additional_passwords[each.key]
terraform_validate ./test/fixtures/mssql-public 
Error: 
Missing resource instance key
  on ../../../modules/mssql/main.tf line 178, in resource "google_sql_user" "additional_users":
 178:   password   = lookup(each.value, "password", 
random_password.additional_passwords
.result)
Because random_password.additional_passwords has "for_each" set, its
attributes must be accessed on specific instances.
For example, to correlate with indices of a referring resource, use:
    random_password.additional_passwords[each.key]
terraform_validate ./test/fixtures/mysql-ha 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/mysql-private 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/mysql-public 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/postgresql-ha 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/postgresql-public 
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/postgresql-public-iam 
Success!
 The configuration is valid.
terraform_validate ./test/setup 
Success!
 The configuration is valid.
 ```

**Comment by samcre** [comment]
@googlebot I signed it!

**Comment by morgante** [comment]
@samcre Please address the formatting issues.

File: modules/mssql/outputs.tf [hunk 1]
```diff
@@ -61,6 +61,17 @@ output "generated_user_password" {
   sensitive   = true
 }
 
+output "additional_users" {
+  description = "Map of additional users and passwords"
+  value = [for r in google_sql_user.additional_users :
```
Comment (morgante) [inline modules/mssql/outputs.tf:66]: This output is a list, not a map (suggest making it a map).

**Review by morgante — CHANGES_REQUESTED** [review]
Thanks for working on this, just one nit.

**Comment by morgante** [comment]
@samcre Looks like the test failure is unrelated, you can ignore it.

File: modules/mssql/outputs.tf (hunk 1 above)
Comment (samcre) [reply modules/mssql/outputs.tf:66]: I updated the output description on every module on the next commit. I did it this way to make it consistent with the input variable `additional_users`, so the input and the outputs have the same format.

**Comment by samcre** [comment]
@morgante Everything is OK? Should I make more changes?

**Review by morgante — APPROVED** [review]

**Comment by tfluehmann** [comment]
HI @samcre 
Will this change overwrite existing passwords?

**Pull request merged**